reporting/performance/test_mode.cpp

```diff
@@ -0,0 +1,39 @@
+//  (C) Copyright Nick Thompson 2020.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <random>
+#include <boost/math/statistics/univariate_statistics.hpp>
+#include <boost/multiprecision/cpp_bin_float.hpp>
+#include <benchmark/benchmark.h>
+
+using boost::multiprecision::cpp_bin_float_50;
+
+template <class Z>
+void test_mode(benchmark::State& state)
+{
+    using boost::math::statistics::mode;
+    
+    std::random_device rd;
+    std::mt19937_64 mt(rd());
+    std::uniform_real_distribution<double> dist {1, 10};
+
+    auto gen = [&dist, &mt](){return dist(mt);};
+
+    std::vector<Z> v(100);
+    std::generate(v.begin(), v.end(), gen);
+
+    for (auto _ : state)
+    {
+        benchmark::DoNotOptimize(mode(v));
+    }
+}
+
+BENCHMARK_TEMPLATE(test_mode, float);
```

Comment:
I feel fairly confident that this should not compile with any floating point type. Otherwise we need to have a semantics for when two floating point numbers are approximately equal.

As it stands, the mode is not robust against perturbations.

Reply:
I added the assertion for integral values with the message that floating point values have not been implemented. I see what you are getting at with the floating point comparison problem.